### PR TITLE
Improve PR testing templates

### PR DESCRIPTION
**Pull request by likawind**
## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,18 +2,28 @@
...
-    - `run_integration_test`: Runs integration tests
-    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)
-
+## Test Plans
+Reviewer note: by approving the PR, reviewer agree that the test plans are solid enough to merge, with risk low enough for daily tests / release eng to take over.
+### Tests specific to this PR
+Include tests specific to this PR's context here:
+### Tests based on components touched by this PR
+Check the components touched by this PR and ensures all items under that component are checked. If certain item is not checked, explicitly state why and agree with PR reviewers.
+- [ ] This PR involves SDK changes
```
Comment (agiron123) [inline .github/pull_request_template.md:11]: Small nit:
"This PR involves SDK changes. I have:
  - included an integration test that covers my change
  - added the 'run_integration_test' label to this PR
  - (if including user-facing APIs) verified that this change is backwards compatible 
  
  etc.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -2,18 +2,28 @@
...
+  - [ ] Have integration test that covers my change
+  - [ ] If changing REST endpoint, have backend test that covers this change (under `integration_tests/backend`)
+  - [ ] Added 'run_integration_test' label to this PR
+  - [ ] This PR involves changes in database layer
+    - [ ] There's a database unit test covering this change
+    - [ ] If the change affects how data is stored, has carefully examined if migration is required.
+- [ ] This PR involves engine-specific changes
+  - [ ] Performed integration tests (manually or automatically) backed by this engine.
+- [ ] This PR involves UI changes
+  - [ ] All manual_qa_tests workflows works with this feature.
```
Comment (kenxu95) [inline .github/pull_request_template.md:25]: It's possible that the manual qa tests don't test this UI feature at all. Should we just add a general note at the bottom that if this is the case we should create a task? Curious what the process is for dealing with manual QA testing gaps.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -2,18 +2,28 @@
...
+  - [ ] Critical integration registration works with this feature. (k8s, s3, and snowflakes) Refer to https://www.notion.so/aqueducthq/Compute-Resource-Setup-Guide-c83e25e1bc6847efbb226f6fc86fa5cd on connecting to integrations.
```
Comment (kenxu95) [inline .github/pull_request_template.md:26]: Can we format this with each of the critical integrations as checkboxes:

- [ ] Critical integration registration works with
    - [ ] K8s
    - [ ] S3
    - [ ] Snowflake (the snowflake form has a lot of fields, can we pick a different one?)

We should include GCP/Bigquery.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -2,18 +2,28 @@
...
+  - [ ] If changing a shared component, ensure all callers to this component are updated.
```
Comment (kenxu95) [inline .github/pull_request_template.md:27]: Curious what the risk is here? If you forget to update a caller correctly won't it fail to build? I do think we should check that all our storybook stories aren't crashing though, as a separate step.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -2,18 +2,28 @@
...
+  - [ ] Included a loom demo / screenshot for this feature.
```
Comment (kenxu95) [inline .github/pull_request_template.md:28]: Should we make a policy that you always either include a loom demo/screenshot or write a test for any new feature changes?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -2,18 +2,28 @@
...
-- [ ] All features on the UI continue to work correctly.
-- [ ] Added one of the following CI labels:
-    - `run_integration_test`: Runs integration tests
-    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)
-
+## Test Plans
+Reviewer note: by approving the PR, reviewer agree that the test plans are solid enough to merge, with risk low enough for daily tests / release eng to take over.
+### Tests specific to this PR
+Include tests specific to this PR's context here:
+### Tests based on components touched by this PR
```
Comment (kenxu95) [inline .github/pull_request_template.md:9]: It seems like there might be steps in here that are only necessary for new feature PRs, and not for refactors, documentation changes etc. Do we have separate checklists for feature development/bug fixes vs NFRs? I know one is probably a superset of the other.

**Review by kenxu95 — COMMENTED** [review]
nit with the tenses: if we're doing present tense, change things like "added" -> "add"

Actually to fix the grammar issues it might be way easier just to run it by ChatGPT :P 

I have some wording nits but will focus on content for this review.

**Review by kenxu95 — COMMENTED** [review]
nit with the tenses: if we're doing present tense, change things like "added" -> "add"

Actually to fix the grammar issues it might be way easier just to run it by ChatGPT :P 

I have some wording nits but will focus on content for this review.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -2,18 +2,28 @@
...
-
-## Checklist before requesting a review
-- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
-- [ ] I have performed a self-review of my code.
-- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
-- [ ] If this is a new feature, I have added unit tests and integration tests.
-- [ ] I have run the integration tests locally and they are passing.
-- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
-- [ ] All features on the UI continue to work correctly.
-- [ ] Added one of the following CI labels:
```
Comment (jpurusho65) [inline .github/pull_request_template.md:15]: We can't add this until review is done right, integration tests before rest of review will consume resources that has cost associated. So description can mention do this once code review is completed.

**Review by saurav-c — COMMENTED** [review]
Echoing sentiment about splitting this into more sections

File: .github/pull_request_template.md (hunk 2 above)
Comment (likawind) [reply .github/pull_request_template.md:25]: That's a good idea, we should ask PR owner to ask manual QA (or guideline) that can help cover the feature. I'd imagine in most cases it should be sufficient to modify an existing workflow

File: .github/pull_request_template.md (hunk 5 above)
Comment (kenxu95) [reply .github/pull_request_template.md:28]: Yeah I'm just saying for all new feature development, regardless of what component it touches. For UI its screenshots but for the others we should always add a new test.

**Pull request closed without merging**